mongo/filterstore.go

```diff
@@ -394,28 +397,30 @@ func selector(filterID, dimensionName string, timestamp bson.MongoTimestamp, eTa
 }
 
 // CreateFilterOutput creates a filter output resource
-func (s *FilterStore) CreateFilterOutput(filter *models.Filter) (err error) {
-	session := s.Session.Copy()
-	defer session.Close()
-
-	filter.UniqueTimestamp, err = bson.NewMongoTimestamp(time.Now(), 1)
+func (s *FilterStore) CreateFilterOutput(ctx context.Context, filter *models.Filter) (err error) {
+	filter.UniqueTimestamp, err = NewMongoTimestamp(time.Now(), 1)
 	if err != nil {
 		return
 	}
 
-	return session.DB(s.db).C(s.outputsCollection).Insert(filter)
+	var documents []interface{}
+	documents = append(documents, filter)
+
+	_, err = s.Connection.C(s.OutputsCollection).Insert(ctx, documents)
```

Comment:
can we use an anonymous array here like: `[]interface{}{filter}`

Reply:
Needs to have key pair values, so would need something in addition to the filter. I'm going to leave as it is as fixing this stuff feels out of scope.